eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -1,5 +1,8 @@
 parameters:
   RunUnitTests: true
+  Artifacts: []
+  ServiceDirectory: not-specified
+  DocArgs: not-specified
```

Comment:
You should try to default DocArgs to be `${{ parameters.ServiceDirectory }}` and then you can probably avoid duplicating it in the yaml files.

Reply:
Azure Devops complains with error - It doesn't like the template args to be used while defining parameters. I'll let this be for now